src/array.ts

```diff
@@ -169,10 +127,10 @@ export class SignalArray<T = unknown> {
         let index = convertToInt(prop);
 
         if (index !== null) {
-          self.#dirtyStorageFor(index);
-          self.#collection.set(null);
```

Comment:
I've been trying to implement this in ember/glimmer, and I wanted to clarify with you -- is this a bug? dirtying the whole collection when changing one item? 

if the index is existing, why would the collection change?